cpp/src/arrow/dataset/scanner.h

```diff
@@ -109,18 +111,34 @@ Status ScanTaskIteratorFromRecordBatch(std::vector<std::shared_ptr<RecordBatch>>
 ///          yield scan_task
 class ARROW_DS_EXPORT Scanner {
  public:
+  Scanner(DataSourceVector sources, std::shared_ptr<ScanOptions> options,
+          std::shared_ptr<ScanContext> context)
+      : sources_(std::move(sources)),
+        options_(std::move(options)),
+        context_(std::move(context)) {}
+
+  virtual ~Scanner() = default;
+
   /// \brief The Scan operator returns a stream of ScanTask. The caller is
   /// responsible to dispatch/schedule said tasks. Tasks should be safe to run
   /// in a concurrent fashion and outlive the iterator.
   virtual ScanTaskIterator Scan() = 0;
 
-  virtual ~Scanner() = default;
-
   /// \brief Convert a Scanner into a Table.
   ///
+  /// \param[in] use_threads guide if the operation will be parallelized
+  ///            with the ThreadPool found in ScanContext.
+  /// \param[out] out output parameter
+  ///
   /// Use this convenience utility with care. This will serially materialize the
   /// Scan result in memory before creating the Table.
-  Status ToTable(std::shared_ptr<Table>* out);
+  Status ToTable(bool use_threads, std::shared_ptr<Table>* out);
```

Comment:
Shouldn't `use_threads` be part of the ScanOptions?

Reply:
I'm not sure if Scanner should have internal nested parallelism, but I'll move it there.